### PR TITLE
(SERVER-2758) Update jruby-utils to 3.1.0

### PR DESCRIPTION
**Pull request by Magisus**
This update also bumps JRuby to 9.2.11.0.

### CODE REVIEW

**Comment by Magisus** [comment]
Still trying to track down the issue where the uberjar built by ezbake does not contain the overridden version of jruby-utils, pulling in the version from clj-parent instead.

File: dev-setup [hunk 1]
```diff
@@ -21,7 +21,7 @@ do
 done < ./resources/ext/build-scripts/jruby-gem-list.txt
 
 echo "Installing vendored gems to '${VENDORED_GEMS_HOME}'"
-lein gem install --install-dir "${VENDORED_GEMS_HOME}" --no-ri --no-rdoc "${gem_list[@]}"
+lein gem install --install-dir "${VENDORED_GEMS_HOME}" --no-document "${gem_list[@]}"
```
Comment (mwaggett) [inline dev-setup]: nitpick, but perhaps these should go in a separate commit.

**Comment by Magisus** [comment]
See last commit for details on how this got around the issue when ezbaking.

**Comment by Magisus** [comment]
Looks like packaging uses `no-ri` and is apparently struggling now?

**Comment by justinstoller** [comment]
Travis is broken
